src/gems/simulation/simulation_table.py

```diff
@@ -20,70 +20,106 @@ class SimulationColumns(str, Enum):
     BASIS_STATUS = "basis-status"
 
 
+class OutputValues:
+    """
+    Minimal typing for OutputValues to satisfy mypy.
+    """
+
+    problem: Any
+    _components: Dict[str, Any]
+    _extra_outputs: Dict[str, Dict[str, ExtraOutput]]
+
+
 class SimulationTableBuilder:
     """Builds simulation tables from solver output values."""
 
     def __init__(self, simulation_id: Optional[str] = None) -> None:
-        self.simulation_id = simulation_id or datetime.now().strftime("%Y%m%d-%H%M")
+        self.simulation_id: str = simulation_id or datetime.now().strftime(
+            "%Y%m%d-%H%M"
+        )
 
     def build(
         self, output_values: OutputValues, absolute_time_offset: Optional[int] = None
     ) -> pd.DataFrame:
-        """Populate a DataFrame from OutputValues."""
         if output_values.problem is None:
             raise ValueError("OutputValues problem is not set.")
 
         context = output_values.problem.context
-        block = context._block.id
+        block = getattr(context._block, "id")
         block_size = context.block_length()
         absolute_time_offset = absolute_time_offset or (block - 1) * block_size
+        assert absolute_time_offset is not None
+        rows: list[dict[str, Any]] = []
```

Comment:
You could even use a union of possible types
```suggestion
        rows: list[dict[str, float | str ...]] = []
```